src/main/java/cubicchunks/world/ICubicWorld.java

```diff
@@ -82,11 +83,12 @@
      * Returns true iff the given Predicate evaluates to true for all cubes for block positions within blockRadius from
      * centerPos. Only cubes that exist are tested. If some cubes within that range aren't loaded - returns false.
      */
-    default boolean testForCubes(BlockPos centerPos, int blockRadius, Predicate<Cube> test) {
+    default boolean testForCubes(BlockPos centerPos, int blockRadius, Predicate<Cube> test, Requirement req) {
```

Comment:
Considering my other comments, I doubt it's needed, and the way Requirement is used is very counterintuitive.

As I explained before, checking if cube is watched can be done in the given predicate.